flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManager.java

```diff
@@ -454,6 +456,11 @@ public void onError(Throwable error) {
 		onFatalError(error);
 	}
 
+	@VisibleForTesting
+	Map<ResourceID, YarnWorkerNode> getWorkerNodeMap() {
+		return workerNodeMap;
```

Comment:
```suggestion
		return Collections.unmodifiableMap(workerNodeMap);
```
We should always be careful exposing non-primitive fields, such as `Map` and `List`. Despite declared `final`, they might still be changed.